backend/azureblob/azureblob.go

```diff
@@ -1838,12 +1844,109 @@ func (o *Object) clearMetaData() {
+		ContentType:                 currentBlob.Properties.ContentType,
+		CopyCompletionTime:          currentBlob.Properties.CopyCompletionTime,
+		CopyID:                      currentBlob.Properties.CopyID,
+		CopyProgress:                currentBlob.Properties.CopyProgress,
+		CopySource:                  currentBlob.Properties.CopySource,
+		CopyStatus:                  currentBlob.Properties.CopyStatus,
+		CopyStatusDescription:       currentBlob.Properties.CopyStatusDescription,
+		CreationTime:                currentBlob.Properties.CreationTime,
+		Date:                        currentBlob.Properties.LastModified,
+		DestinationSnapshot:         currentBlob.Properties.DestinationSnapshot,
+		EncryptionKeySHA256:         currentBlob.Properties.CustomerProvidedKeySHA256,
+		EncryptionScope:             currentBlob.Properties.EncryptionScope,
+		ETag:                        currentBlob.Properties.ETag,
+		ExpiresOn:                   currentBlob.Properties.ExpiresOn,
+		ImmutabilityPolicyExpiresOn: currentBlob.Properties.ImmutabilityPolicyExpiresOn,
+		ImmutabilityPolicyMode:      currentBlob.Properties.ImmutabilityPolicyMode,
+		IsCurrentVersion:            &isCurrentVersion,
+		IsIncrementalCopy:           currentBlob.Properties.IncrementalCopy,
+		IsSealed:                    currentBlob.Properties.IsSealed,
+		IsServerEncrypted:           currentBlob.Properties.ServerEncrypted,
+		LastAccessed:                currentBlob.Properties.LastAccessedOn,
+		LastModified:                currentBlob.Properties.LastModified,
+		LeaseDuration:               currentBlob.Properties.LeaseDuration,
+		LeaseState:                  currentBlob.Properties.LeaseState,
+		LeaseStatus:                 currentBlob.Properties.LeaseStatus,
+		LegalHold:                   currentBlob.Properties.LegalHold,
+		Metadata:                    currentBlob.Metadata,
+		ObjectReplicationPolicyID:   nil,
+		ObjectReplicationRules:      nil,
+		RehydratePriority:           (*string)(currentBlob.Properties.RehydratePriority),
+		RequestID:                   nil,
+		TagCount:                    &tagCount,
+		Version:                     nil,
+		VersionID:                   nil,
+	}
+}
+
+// readMetaData gets the metadata using a ListBlobsFlatPager, to avoid requiring read access on the blob itself, and is used by Fs.readMetaData if no_read_for_metadata is set
+func (f *Fs) readMetadataUsingList(ctx context.Context, containerName, containerPath string) (blobProperties blob.GetPropertiesResponse, err error) {
+	blobsPager := f.cntSVC(containerName).NewListBlobsFlatPager(&container.ListBlobsFlatOptions{
+		Include: container.ListBlobsInclude{
+			Metadata: true,
+		},
+		Prefix: &containerPath,
+	})
+	for blobsPager.More() {
+		blobs, err := blobsPager.NextPage(ctx)
+		if err != nil {
+			return blobProperties, err
+		}
+
+		for _, currentBlob := range blobs.ListBlobsFlatSegmentResponse.Segment.BlobItems {
+
+			if *currentBlob.Name != containerPath {
```

Comment:
This needs a bit of Encoding - `*currentBlob.Name` will be in backend encoding where `containerPath` is in Standard encoding I think.